app/response/views.py

```diff
@@ -91,32 +92,25 @@ def response_file(request_id):
     Call process_upload_data to process the uploaded file form data.
     Pass data into helper function in response.utils to update changes into database.
     Send email notification to requester and bcc agency users if privacy is release.
-    Render specific tempalte and send email notification bcc agency users if privacy is private.
+    Render specific template and send email notification bcc agency users if privacy is private.
 
     :param request_id: FOIL request ID for the specific file.
     :return: redirect to view request page
     """
     current_request = Requests.query.filter_by(id=request_id).first()
     files = process_upload_data(flask_request.form)
+    agency_file_links = dict()
+    agency_file_links['private'] = dict()
+    agency_file_links['release'] = dict()
+    requester_file_links = dict()
     for file_data in files:
-        add_file(current_request.id,
-                 file_data,
-                 files[file_data]['title'],
-                 files[file_data]['privacy'])
-    file_options = process_privacy_options(files)
-    email_content = flask_request.form['email-file-summary']
-    for privacy, files in file_options.items():
-        if privacy == PRIVATE:
-            send_file_email(request_id,
-                            privacy,
-                            files,
-                            None,
-                            email_template='email_templates/email_private_file_upload.html')
-        else:
-            send_file_email(request_id,
-                            privacy,
-                            files,
-                            email_content)
+        response_obj = add_file(current_request.id,
+                                file_data,
+                                files[file_data]['title'],
+                                files[file_data]['privacy'])
+        agency_file_links, requester_file_links = get_file_links(response_obj, agency_file_links, requester_file_links)
```

Comment:
You are passing dictionaries into `get_file_links` that you do not use (in the scope of `response_file`) until after `get_file_links` returns so why aren't you generating them inside `get_file_links`?
One other thing: dictionaries are *mutable*. So, if `get_file_links` is returning "updated versions" of the file_links dicts without rebinding them, there is really no need to return them at all.

Test this out for yourself:
```python
def update_dict(dictionary):
    dictionary['baz'] = 'qux'
    # does not return the dictionary

d = {'foo': 'bar'}
update_dict(d)
print(d)

def rebind_dict(dictionary):
    dictionary = {'quux': 'quuz'}

rebind_dict(d)
print(d)
```

For more info, google passing by reference/value in python 
and/or check out: https://jeffknupp.com/blog/2012/11/13/is-python-callbyvalue-or-callbyreference-neither/